### DISCUSSION TITLE
Adding Certificate of fqdn in RK2 config file

### DISCUSSION THREAD

**Discussion by cloudcafetech** [rancher/rke2#1842] [Q&A]
How to add certificate (pem) of fqdn (LB) in RK2 config ?

![image](https://user-images.githubusercontent.com/68260844/132933973-330dc8b3-072d-4213-9b70-5bb8f7403129.png)

**Comment by cloudcafetech** [comment]
Any reason for closing the issue?

**Reply by brandond** [reply]
This is better handled as a discussion question since nothing is broken. Also, it seems like it belongs in the Rancher repo?

**Reply by cloudcafetech** [reply]
but If this is answered then I think we can solve https://github.com/rancher/rke2/issues/1831 (please check my last statement)

**Reply by cloudcafetech** [reply]
Is there any way we can add certificate (pem) of LB in RK2 config ?

**Reply by brandond** [reply]
I'm not even sure what that text box is for. Without involving Rancher, what in RKE2 itself are you trying to do?

**Reply by cloudcafetech** [reply]
If you add certificate in caCert then we can resolve the issue #1831

**Reply by brandond** [reply]
I think you're talking about the question you asked in the other thread, which was
> I think that can solve if you tell me how to add certificate (pem) of fqdn (LB) in RK2 config file ... how normally we do in GUI

Unfortunately I don't know what config file you're talking about, nor do I recognize the fragment of GUI that you've screenshotted.

Can you explain in more depth what your end goal is? Are you trying to connect to the RKE2 apiserver when it's behind a load-balancer that's terminating SSL? If so, as I said earlier - you need to turn off SSL termination on your load-balancer if you want to use the admin kubeconfig generated by RKE2 on startup; this is not a viable configuration. In addition to the load-balancer not using the certificate specified by the kubeconfig, SSL terminating load-balancers also prevent the apiserver from seeing the client certificate that the Kubernetes client uses to authenticate itself.

**Reply by cloudcafetech** [reply]
OK, just help me to understand how rke2 create certificate ?
